ingestion/src/metadata/ingestion/lineage/sql_lineage.py

```diff
@@ -64,20 +65,33 @@ def search_table_entities(
     if search_tuple in search_cache:
         return search_cache.get(search_tuple)
     try:
-        table_fqns = fqn.build(
-            metadata,
+        table_entities: Optional[List[Table]] = []
+        # search on ES first
+        fqn_search_string = build_es_fqn_search_string(
+            database, database_schema, service_name, table
+        )
+        es_result_entities = metadata.es_search_from_fqn(
```

Comment:
Are we doing that elasticsearch call as it is less expensive than the actual OM Server?

Reply:
When calling `fqn.build`, we already get the Table entities from ES, so we are doing an unnecessary extra call to the API with the FQN. So the idea in this PR is to use the result of ES directly; otherwise, call to the API using the `get_by_name` method with the FQN.